src/crawlers/browser_crawler.js

```diff
@@ -414,8 +436,13 @@ export default class BrowserCrawler extends BasicCrawler {
         const browserControllerInstance = this.browserPool.getBrowserControllerByPage(page);
         crawlingContext.browserController = browserControllerInstance;
 
-        crawlingContext.session = browserControllerInstance.launchContext.session;
-        crawlingContext.proxyInfo = browserControllerInstance.launchContext.proxyInfo;
+        if (!crawlingContext.session) {
+            crawlingContext.session = browserControllerInstance.launchContext.session;
```

Comment:
There will always be a `session` because `BasicCrawler` creates it for all requests. So this will always be `true`.

https://github.com/apify/apify-js/blob/77626db26f7d9627116373596aabbc29c3a01da0/src/crawlers/basic_crawler.js#L498-L504

Reply:
Indeed it fixed the failing test. However, this is still a mess. Sessions should be set only at one place. Now it's just scattered around and tough to play with.

Reply:
Hmm... I was thinking of another crawler option. Instead of `launchContext.useIncognitoPages`, why not just `reuseSessions`? The behavior should be the same no matter what crawler is used.

Reply:
The thing is `reuseSessions` or similar sounds like an option of the `SessionPool`, not the crawler. The crawlers should just consume whatever `SessionPool` gives them. With that said, `SessionPool` is definitely not ideal. It helped us a lot, but now I think we have more experience that we should leverage to create a new one.

I coined it `UserPool` because it would generate users, with various "fingerprints" and then rotate the users based on some strategy. It's only in an idea stage now, but I think that if we decide that this logic should be changed, we should first think about `UserPool` and how to make it as helpful as possible for the devs.

Reply:
> The thing is reuseSessions or similar sounds like an option of the SessionPool, not the crawler.

Yeah, I was thinking it would be passed along. It just seems weird that it's in `launchContext`.